app/components/SwitchSetting.js

```diff
@@ -0,0 +1,70 @@
+import { Platform, StyleSheet, View } from "react-native";
+import { Text } from "./index";
+import { Switch } from "@rneui/themed";
+import React, { useContext } from "react";
+import { ThemeContext } from "../theme-context";
+
+const SwitchSetting = ({ title, description, value, onValueChange }) => {
+  const { theme } = useContext(ThemeContext);
+  const s = getStyles(theme);
+
+  return (
+    <View>
+      <View style={[s.title, { marginBottom: 4 }]}>
+        <Text style={s.titleText}>{title}</Text>
+      </View>
+      <View
+        style={{
+          flexDirection: "row",
+          alignItems: "center",
+          marginVertical: -8,
+        }}
+      >
+        <Text style={[s.descriptionText, { flex: 1 }]}>{description}</Text>
+        <Switch
+          value={value}
+          onValueChange={() => {
+            // Sends the new checked value
+            return onValueChange(!value);
+          }}
+          trackColor={s.switchTrackColor}
+          thumbColor={
+            value ? s.switchThumbColors.true : s.switchThumbColors.false
+          }
```

Comment:
I'm not sure what the logic is for determining which flavor of track color to use, but the reference currently as `s.switchTrackColor` will not apply any styles. 

For thumb color:
```suggestion
          thumbColor={value ? theme.white : theme.base3}
```

Reply:
I removed the thumb color, since the default one seems fine regardless of theme.

Seems like the color for the track uses ios_backgroundColor on iOS when the value is false (https://reactnative.dev/docs/switch#ios_backgroundcolor-ios). And the true style of trackColor when it's true.

Looks like Android do use the false in the trackColor style when the value is false.